crates/plugins/src/version_check.rs

```diff
@@ -0,0 +1,67 @@
+use crate::{
+    get_manifest_file_name, plugin_manifest::PluginManifest, PLUGIN_MANIFESTS_DIRECTORY_NAME,
+};
+use anyhow::{anyhow, Result};
+use semver::{Version, VersionReq};
+use std::{fs::File, path::Path};
+
+/// Checks whether the plugin supports the currently running version of Spin.
+pub fn assert_supported_version(spin_version: &str, supported: &str) -> Result<()> {
+    let supported = VersionReq::parse(supported).map_err(|e| {
+        anyhow!(
+            "could not parse manifest compatibility version {} as valid semver -- {:?}",
+            supported,
+            e
+        )
+    })?;
+    let version = Version::parse(spin_version)?;
+    match supported.matches(&version) {
+        true => Ok(()),
+        false => Err(anyhow!(
+            "plugin is compatible with Spin {} but running Spin {}",
+            supported,
+            spin_version
+        )),
```

Comment:
Again, is there or should there be a way for the user to go YOLO SHIP IT ANYWAY?

Generally, should we distinguish between "are you really sure you want to do this" warnings (the author hasn't updated their version spec yet) vs "we can't do this at all" (the upstream site is gone) errors?